src/util/googleMaps.js

```diff
@@ -154,3 +154,141 @@ export const locationBounds = (latlng, distance) => {
+ * @param {object} Type - Google Maps class/type
+ * @param {function} factory - function to create instance of given type from object literal version
+ *
+ * @return an instance of given Type
+ */
+const ensureOfType = (inst, type, factory) => {
+  if (inst instanceof type) {
+    return inst;
+  } else {
+    return factory(inst, type);
+  }
+};
+
+/**
+ * Helper functions for handling Google OverlayView
+ * Based on https://github.com/tomchentw/react-google-maps/blob/master/src/utils/OverlayViewHelper.js
+ *
+ * @param {MapCanvasProjection} mapCanvasProjection - map projection.
+ * @param {object} offset - position offset on map canvas.
+ * @param {LatLngBounds} bounds - map bounds.
+ *
+ * @return styles to render the overlay within the projection.
+ */
+const getLayoutStylesByBounds = (mapCanvasProjection, offset, bounds) => {
+  const ne = mapCanvasProjection.fromLatLngToDivPixel(bounds.getNorthEast());
+  const sw = mapCanvasProjection.fromLatLngToDivPixel(bounds.getSouthWest());
+  if (ne && sw) {
+    return {
+      left: `${sw.x + offset.x}px`,
+      top: `${ne.y + offset.y}px`,
+      width: `${ne.x - sw.x - offset.x}px`,
+      height: `${sw.y - ne.y - offset.y}px`,
+    };
+  }
+  return {
+    left: `-9999px`,
+    top: `-9999px`,
+  };
+};
+
+/**
+ * Helper functions for handling Google OverlayView
+ * Based on https://github.com/tomchentw/react-google-maps/blob/master/src/utils/OverlayViewHelper.js
+ *
+ * @param {MapCanvasProjection} mapCanvasProjection - map projection.
+ * @param {object} offset - position offset on map canvas.
+ * @param {LatLng} position - map position/location.
+ *
+ * @return  styles to render single coordinate pair within the projection.
+ */
+const getLayoutStylesByPosition = (mapCanvasProjection, offset, position) => {
+  const point = mapCanvasProjection.fromLatLngToDivPixel(position);
+  if (point) {
+    const { x, y } = point;
+    return {
+      left: `${x + offset.x}px`,
+      top: `${y + offset.y}px`,
+    };
+  }
+  return {
+    left: `-9999px`,
+    top: `-9999px`,
+  };
+};
+
+/**
+ * Helper functions for handling Google OverlayView
+ * Based on https://github.com/tomchentw/react-google-maps/blob/master/src/utils/OverlayViewHelper.js
+ *
+ * @param {MapCanvasProjection} mapCanvasProjection - map projection.
+ * @param {object} offset - position offset on map canvas.
+ * @param {object} props - map props.
+ *
+ * @return styles to render an area or a single coordinate pair within the projection.
+ */
+export const getLayoutStyles = (mapCanvasProjection, offset, props) => {
+  if (props.bounds) {
+    const bounds = ensureOfType(props.bounds, google.maps.LatLngBounds, createLatLngBounds);
+    return getLayoutStylesByBounds(mapCanvasProjection, offset, bounds);
+  } else {
+    const position = ensureOfType(props.position, google.maps.LatLng, createLatLng);
+    return getLayoutStylesByPosition(mapCanvasProjection, offset, position);
+  }
+};
```

Comment:
I think the helper functions in this file are very specific, and only internal helpers for this single export. Could they be internal functions here to avoid too much confusion?

You should probably also link directly to a commit hash version in GitHub, not just the master version of the file. That link might not contain anything in the future.

Reply:
@kpuputti, could you elaborate more? There are just 2 new exports: `getOffsetOverride` & `getLayoutStyles`. Are you suggesting that I move those internal functions inside of the scope of `getLayoutStyles` function (or something else)?

Reply:
I mean getting rid of functions like `createLatLng`, `createLatLngBounds`, and `ensureOfType` that are only used in this export might make sense. They are not that useful as generic helpers.

Reply:
I'll move those inside `getLayoutStyles`, but I don't want to completely remove them since these are rarely-touched helpers - and I think they should be easy to map with functions found on the other side of the reference link to react-google-maps repo.

And I changed those link to point to the latest tag instead of master.